### ISSUE TITLE
Improve clarity on the limitation of using one kubeconfig file for multiple clusters

### ISSUE THREAD

**Issue by guettli** [kubernetes/website#40944]
I think the page should be more explicit about the fact that using one kube config file has a major drawback.

https://kubernetes.io/docs/tasks/access-application-cluster/configure-access-multiple-clusters/

Imagine you want to connect to cluster A and cluster B.

You have two shells. One for cluster A, one for cluster B.

If you use `kubectl config use-context <context-name>`, then the context changes for **both** shells.

If you use two kube config files and set KUBECONFIG, then you can easily work with two shells/terminals and two clusters.

Tools like https://direnv.net help via .envrc files which set env vars if you enter the directory like this: `cd directory-of-cluster-a`.

**Comment by dipesh-rawat** [comment]
Page Related to Issue: https://kubernetes.io/docs/tasks/access-application-cluster/configure-access-multiple-clusters/
/language en

**Comment by sftim** [comment]
@guettli could you mention the URL of the page within the issue description itself? That'd help.

We could also, if we want, talk about making shell helpers.

For example: alias kubectl to `kctl-prod` and `kctl-dev` if you have two different clusters and use a different kubeconfig (or context) for each.
It's important to let readers know that There Is More Than One Way To Do It.

**Comment by guettli** [comment]
@sftim yes, there are several ways. But one think should be documented: `kubectl use-context` fails if you want to use several clusters in parallel. 

(and I added the URL to the page (see abover)).

**Comment by sftim** [comment]
> `kubectl use-context` fails if you want to use several clusters in parallel.

If you mean `kubectl config use-context`, that works fine with several different clusters. I do this and it works how I want it to. I don't think we have something to fix in that respect.

The problem is wanting to use two different _shell sessions_, not two different clusters. This is not a detail of having two contexts in the same cluster vs. two contexts in different clusters; it is more about knowing how to use kubectl and your shell.

The shell I use, bash, has features that let me call kubectl in parallel in different shell sessions, where those sessions refer to different clusters, and this works whether I have a combined kubeconfig or separate files. Not all shells have these features; however, that's not really a shortcoming in kubectl.

We could provide a tutorial on doing this; that would be a feature request rather than a bug.

Given that:
/kind support

**Comment by guettli** [comment]
@sftim could you please share your knowledge how to have one combined kubeconfig and two parallel shells (to two different clusters)?

**Comment by sftim** [comment]
> @sftim could you please share your knowledge how to have one combined kubeconfig and two parallel shells (to two different clusters)? I use the bash shell, too.

Here's one approach:
```bash
# shell 1
alias kubeconfig='kubeconfig --context foo'
```

```bash
# shell 2
alias kubeconfig='kubeconfig --context bar'
``` 

You can write a script that sets these aliases based on various things. That script would be out of scope IMO.

**Comment by guettli** [comment]
I asked Google and ChatGPT, both don't know the command `kubeconfig`. Is this a typo and you mean `kubectl`?

**Comment by sftim** [comment]
Ah yep, I was typing quickly. Sorry about mixing that up.

**Comment by sftim** [comment]
I propose to close this issue. We could have a blog article but otherwise there's not really anything to document about Kubernetes itself.
